### DISCUSSION TITLE
Team with 'educationClass' template

### DISCUSSION THREAD

**Discussion by Maki079** [microsoftgraph/msgraph-sdk-go#663] [Q&A]
Hi !
I'm asking for your help because I'm loosing my mind at this point.
I want to create a new Team using 'educationClass' template but i need it to NOT be activated at start, so owners can prepare the classes before students can se it. 
This can be easily done using  MicrosoftTeams PowerShell Gallery Module ![ref](https://www.powershellgallery.com/packages/MicrosoftTeams/5.9.0)
like this : 

New-Team -DisplayName "Super Team 420" -MailNickName "SuperTeam" -Template "EDU_Class" -Owner (owner@example.com")

BUT calling Graph ends with a team witch is already activated (it is impossible to "unactive" a team) 

```go
        requestBody := graphmodels.NewTeam()

	requestBody.SetDisplayName(&teamDisplayName)

	requestBody.SetDescription(&teamDescription)

	isMembershipLimitedToOwners := true

	requestBody.SetIsMembershipLimitedToOwners(&isMembershipLimitedToOwners)

	conversationMember := graphmodels.NewAadUserConversationMember()
	roles := []string{
		"owner",
	}
	conversationMember.SetRoles(roles)
	additionalOwnerData := map[string]interface{}{
		"user@odata.bind": "https://graph.microsoft.com/v1.0/users('" + teamOwnerId + "')",
	}
	conversationMember.SetAdditionalData(additionalOwnerData)

	members := []graphmodels.ConversationMemberable{
		conversationMember,
	}
	requestBody.SetMembers(members)

	additionalData := map[string]interface{}{
		"template@odata.bind": "https://graph.microsoft.com/v1.0/teamsTemplates('educationClass')",
	}
	requestBody.SetAdditionalData(additionalData)
```

Have anyone done something like this before ? I`m extremely thankful for any response

**Comment by baywet** [comment]
Hi @Maki079 
This repository and its discussions are dedicated to the Go SDK itself, service teams are not watching it.
I recommend you instead:
1. get the JSON representation of the serialized object the Go SDK is sending (see below)
2. open a discussion on the dedicated [Microsoft Q&A forum](https://learn.microsoft.com/en-us/answers/tags/282/microsoft-graph-teamwork-api/) where the service team will be able to answer you.

I suspect the Microsoft Teams PowerShell module is either tweaking the payload in a specific way, or using some non-public backend administrative API, hence the difference in behaviour.

```go
bytes, err := abstractions.SerializeToJson(requestBody)
jsonRepresentation := string(bytes)
```

https://github.com/microsoft/kiota-abstractions-go/blob/2291b081d5b2e6337192f4925678ed0fd7874de6/serialization/kiota_json_serializer.go#L6

**Comment by daedalus1776** [comment]
@Maki079 I realize this isn't the right place, but did you ever figure this out? I'm in the same boat and need these teams not activated. Using the MicrosoftTeams module causes weird issues on Edu_Class templates (such as onenote not working)

**Reply by Maki079** [reply]
Hi!
I managed to figure it out, it's a "pain in the ass" solution but it works XD. 
You need to first create Education Class (its a group type)

`classes, err := graphClient.Education().Classes().Post(context.Background(), requestBody, nil)`

Then add to this group at least 1 teacher (aka owner) and preferably also students (i found it faster to add users to group rather than Team).

Next is to create Team out of Education Class :

`
team, err := graphClient.Teams().Post(context.Background(), requestBody, nil)
`
But in the request body you need to set those parameters :

```
requestBody := graphmodels.NewTeam()

	isMembershipLimitedToOwners := true

	requestBody.SetIsMembershipLimitedToOwners(&isMembershipLimitedToOwners)

	additionalData := map[string]interface{}{
		"template@odata.bind": "https://graph.microsoft.com/beta/teamsTemplates('educationClass')",
		"group@odata.bind":    "https://graph.microsoft.com/beta/groups('" + educationClassId + "')",
	}

	requestBody.SetAdditionalData(additionalData)
```

You will end up with Team that is not activated and the result is similar to what this powereshell module created. (i do not have a 100% confirmation if it is the same steps that Microsoft took in theirs implementation )
